spec/frontend.test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import $ from 'jquery';
+
+// Components
+import PhotoBanner from '../client/components/PhotoBanner';
+import PhotoGrid from '../client/components/PhotoGrid';
+
+describe('PhotoBanner Test Suite', () => {
+  test('initial state of PhotoBanner', () => {
+    const wrapper = mount(<PhotoBanner />);
+    expect(wrapper.state('roomId')).toBe('5' || '10');
+    expect(wrapper.state('photos')).toHaveLength(0);
+  });
+
+  test('rendering of PhotoGrid upon available photos', () => {
+    const wrapper = mount(<PhotoBanner />);
+    const photos = [
+      { url: 'https://japaneseanimeinfo.up.seesaa.net/image/401-3e556.jpg' },
+      { url: 'https://i.pinimg.com/474x/47/14/f2/4714f2e99aef38240636d17e976f1672.jpg' },
+    ];
+
+    expect(wrapper.state('photos')).toHaveLength(0);
+    expect(wrapper.find('PhotoGrid')).toHaveLength(0);
+    wrapper.setState({ photos });
+    expect(wrapper.state('photos')).toHaveLength(2);
+    expect(wrapper.find('PhotoGrid')).toHaveLength(1);
+  });
+
+  // In-progress spy
+  test('links trigger an ajax call to getPhotosById', () => {
+    const wrapper = shallow(<PhotoBanner />);
+    const spy = jest.spyOn($, 'ajax');
+    wrapper.instance().getPhotosById();
+    expect(spy).toBeCalledWith({
+      method: 'GET',
+      url: '/api/photos/5',
+    });
+    expect(spy).toHaveBeenCalled();
+    spy.mockRestore();
```

Comment:
Interesting. I was attempting to access the method that contained my API call, not to spy on the caller itself.

Reply:
While looking for a means to access the method, I came across spy. I'll want to return to this test case and investigate more to see how to improve this.